app/src/interfaces/_system/system-filter/system-filter.vue

```diff
@@ -0,0 +1,236 @@
+		const { t } = useI18n();
+		const fieldsStore = useFieldsStore();
+
+		watch(
+			() => props.value,
+			(newVal) => {
+				if (newVal === null) {
+					innerValue.value = [];
+					return;
+				}
+				const newFilter = constructFilterTree(newVal);
+				if (isEqual(innerValue.value, newFilter) === false && newFilter.type === 'logic')
+					innerValue.value = newFilter.values;
+			},
+			{ immediate: true }
+		);
+
+		watch(
+			innerValue,
+			debounce((newVal: FilterTree) => {
+				const newFilter = deconstructFilterTree({
+					type: 'logic',
+					name: '_and',
+					values: newVal,
+				});
+				if (isEqual(newFilter, props.value) === false) emit('input', newFilter);
+			}, 200),
+			{ deep: true }
+		);
+
+		return { t, innerValue, deconstructFilterTree, constructFilterTree, addNode, removeNode };
+
+		function addNode(type: 'logic' | 'field') {
+			const list = innerValue.value;
+
+			if (type === 'logic') {
+				list.push({
+					type: 'logic',
+					name: '_and',
+					values: [],
+				});
+			} else {
+				list.push({
+					type: 'field',
+					name: fieldsStore.getPrimaryKeyFieldForCollection(props.collectionName).field,
+					comparator: '_eq',
+					value: 0,
+				});
+			}
+			innerValue.value = list;
+		}
+
+		function removeNode(ids: number[]) {
+			const id = ids.pop();
+			if (ids.length === 0) {
+				innerValue.value = innerValue.value.filter((node, index) => index !== id);
+				return;
+			}
+			let list = get(innerValue.value, idsToPath(ids)) as FilterTree;
+
+			list = list.filter((node, index) => index !== id);
+
+			innerValue.value = set(innerValue.value, idsToPath(ids), list);
+		}
+
+		function idsToPath(id: number[]) {
+			return id.map((v) => `[${v}]`).join('.values.') + '.values';
+		}
+
+		function deconstructFilterTree(tree: FilterNode): Record<string, any> | null {
+			if (tree === undefined) return null;
+			if (tree.type === 'logic') {
+				return {
+					[tree.name]: tree.values.map((value) => deconstructFilterTree(value)),
+				};
+			} else {
+				if (!tree.name) return null;
+				return set({}, tree.name, {
+					[tree.comparator]: tree.value,
+				});
+			}
+		}
+
+		function constructFilterTree(tree: Record<string, any>, id = [0]): FilterNode {
+			const key = Object.keys(tree)[0];
+			if (key.startsWith('_') && Array.isArray(tree[key]) && key) {
+				return {
+					type: 'logic',
+					name: key as LogicOperators,
+					values: (tree[key] as Record<string, any>[]).map((subTree, index) =>
+						constructFilterTree(subTree, [...id, index])
+					),
+				};
+			} else {
+				let destructTree = tree;
+				let fieldList = [];
+				let key = Object.keys(destructTree)[0];
+
+				while (key.startsWith('_') === false) {
+					fieldList.push(key);
+					destructTree = destructTree[key];
+					key = Object.keys(destructTree)[0];
+				}
+
+				return {
+					type: 'field',
+					name: fieldList.join('.'),
+					comparator: key as FilterOperators,
+					value: destructTree[key],
+				};
+			}
+		}
+	},
```

Comment:
Lets make this interface use the regular (documented) filter structure, instead of converting to/from a proprietary array-based structure.